### PR TITLE
feat: added option for systemd-timesyncd

### PR DESCRIPTION
**Pull request by builderjer**
Companion to https://github.com/OpenVoiceOS/ovos-utils/pull/200

### CODE REVIEW

File: ovos_PHAL_plugin_system/__init__.py [hunk 1]
```diff
@@ -210,16 +210,23 @@ def handle_ssh_disable_request(self, message):
...
-            page = join(dirname(__file__), "ui", "Status.qml")
-            self.gui["status"] = "Enabled"
-            self.gui["label"] = "Clock updated"
-            self.gui.show_page(page)
-        self.bus.emit(message.reply('system.ntp.sync.complete'))
+        # Check to see what service is installed
+        if check_service_installed('ntp'):
+            ntp_sync()
+        elif check_service_installed('systemd-timesyncd')
+            timesync_sync()
```
Comment (JarbasAl) [inline ovos_PHAL_plugin_system/__init__.py:217]: go we move both these methods from ovos utils to this plugin and deprecate them in ovos-utils?

File: ovos_PHAL_plugin_system/__init__.py [hunk 2]
```diff
@@ -273,7 +302,31 @@ def handle_mycroft_restart_request(self, message):
             page = join(dirname(__file__), "ui", "Restart.qml")
             self.gui.show_page(page, override_animations=True,
                                override_idle=True)
-        restart_service(self.service_name, sudo=self.use_root)
+        user = False
+        sudo = False
+        # NOTE: Can self.core_service_name be used here?
```
Comment (JarbasAl) [inline ovos_PHAL_plugin_system/__init__.py:307]: yes do this change please

File: ovos_PHAL_plugin_system/__init__.py [hunk 3]
```diff
@@ -7,28 +7,35 @@
...
         """ this method is called before loading the plugin.
         If it returns False the plugin is not loaded.
         This allows a plugin to run platform checks"""
+        # check if admin plugin is not enabled
+        cfg = Configuration().get("PHAL", {}).get("admin", {})
+        if cfg.get("ovos-PHAL-plugin-system", {}).get("enabled"):
+            # run this plugin in admin mode (as root)
+            return False
+
+        # check if nmcli is installed, assume polkit allows non-root usage
```
Comment (JarbasAl) [inline ovos_PHAL_plugin_system/__init__.py:37]: update or remove this comment leftover

**Comment by JarbasAl** [comment]
can we add a use_sudo flag for all those commands? this works if sudo is passwordless, would be set in mycroft.conf

File: ovos_PHAL_plugin_system/__init__.py (hunk 2 above)
Comment (builderjer) [reply ovos_PHAL_plugin_system/__init__.py:307]: Can this detect if it is a user service or a system service?

**Comment by builderjer** [comment]
> can we add a use_sudo flag for all those commands? this works if sudo is passwordless, would be set in mycroft.conf

Not sure what you mean here.  Where would this be added?

**Comment by j1nx** [comment]
Gentle bump.

is this one still valid? Will see if i can implement this PR for testing for the buildroot image.

**Comment by builderjer** [comment]
I think there's a typo somewhere,  but use it works afik

**Comment by builderjer** [comment]
@j1nx can you look again. It is loading well now

**Review by JarbasAl — APPROVED** [review]

**Comment by neatbasis** [comment]
> > @j1nx can you look again. It is loading well now
> 
> As all PR's have been merged related to this, the code will be present in the nightly alpha containers so will have a look next time I boot a new build.
> 
> Thanks for this as it was one of my PHAL issues to resolve.

Are you sure it's not broken without my PR?

**Comment by j1nx** [comment]
> Are you sure it's not broken without my PR?

Haven't tested it as of yet. Will have a look later but guess by te time I get to it your PR is also merged, so all will be good.

**Pull request merged**